### PR TITLE
Update to Gradle 6.1.1, Android Gradle plugin 4.0.1, Kotlin 1.4 and Coroutines 1.3.9.

### PR DESCRIPTION
**Pull request by pocmo**

### CODE REVIEW

**Comment by pocmo** [comment]
> [task 2020-09-28T14:57:59.545Z] Execution failed for task ':browser-engine-gecko-beta:packageDebugAndroidTest'.
[task 2020-09-28T14:57:59.545Z] > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
[task 2020-09-28T14:57:59.545Z]    > java.lang.OutOfMemoryError (no error message)

😭

**Comment by pocmo** [comment]
I tried to raise the limit but got:

```
[task 2020-09-28T16:48:18.695Z] # There is insufficient memory for the Java Runtime Environment to continue.
[task 2020-09-28T16:48:18.695Z] # Native memory allocation (mmap) failed to map 57671680 bytes for committing reserved memory.
[task 2020-09-28T16:48:18.695Z] OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x0000000776980000, 57671680, 0) failed; error='Cannot allocate memory' (errno=12)
[task 2020-09-28T16:48:18.695Z] # An error report file with more information is saved as:
[task 2020-09-28T16:48:18.695Z] # /builds/worker/checkouts/src/hs_err_pid4079.log
[task 2020-09-28T16:48:19.751Z] 
[task 2020-09-28T16:48:19.790Z] > Task :ktlint FAILED
[task 2020-09-28T16:48:22.646Z] > Task :feature-containers:lintRelease
[task 2020-09-28T16:48:24.246Z] > Task :feature-accounts:lintRelease
[task 2020-09-28T16:48:24.546Z] > Task :feature-accounts-push:lintRelease
[task 2020-09-28T16:48:25.446Z] > Task :feature-app-links:lintRelease
[task 2020-09-28T16:48:26.546Z] > Task :feature-awesomebar:lintRelease
[task 2020-09-28T16:48:28.846Z] > Task :concept-engine:lintRelease
[task 2020-09-28T16:48:29.246Z] > Task :feature-contextmenu:lintRelease
[task 2020-09-28T16:48:29.546Z] 
[task 2020-09-28T16:48:29.546Z] FAILURE: Build failed with an exception.
[task 2020-09-28T16:48:29.546Z] 
[task 2020-09-28T16:48:29.546Z] * What went wrong:
[task 2020-09-28T16:48:29.546Z] Execution failed for task ':ktlint'.
[task 2020-09-28T16:48:29.546Z] > Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 1
```

Looks like this may have hit a physical limit this time?

**Review by rocketsroger — APPROVED** [review]

File: components/feature/addons/src/main/res/layout/mozac_feature_addons_item.xml [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
     <androidx.appcompat.widget.AppCompatImageView
         android:id="@+id/add_on_icon"
+        android:layout_width="@dimen/mozac_widget_favicon_size"
+        android:layout_height="@dimen/mozac_widget_favicon_size"
         style="@style/Mozac.Widgets.Favicon"
```
Comment (Amejia481) [inline components/feature/addons/src/main/res/layout/mozac_feature_addons_item.xml:21]: lint issue? That's unfortunate as this way we will have to update `width` and `height`, when these values change on `Mozac.Widgets.Favicon`

File: components/tooling/detekt/build.gradle [hunk 2]
```diff
@@ -17,6 +17,13 @@ dependencies {
     testImplementation Dependencies.tools_detekt_test
 }
 
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+    // Gradle keeps warning about different Kotlin versions being on the classpath. I am unsure
+    // how to resolve that and if it is even resolvable in this cross-library setup. Anyhow,
+    // since it is only a warning I am going to suppress them for this project specifically.
+    kotlinOptions.allWarningsAsErrors = false
```
Comment (Amejia481) [inline components/tooling/detekt/build.gradle:24]: Should we file an issue for this? Although `allWarningsAsErrors` were completely annoying for me in development there are helpful to catch early possible bugs.

**Review by Amejia481 — APPROVED** [review]

File: components/tooling/detekt/build.gradle (hunk 2 above)
Comment (pocmo) [reply components/tooling/detekt/build.gradle:24]: They are still enabled everywhere else. This is here is for `tooling-detekt`.

Comment (Amejia481) [reply components/tooling/detekt/build.gradle:24]: 👍

File: components/feature/addons/src/main/res/layout/mozac_feature_addons_item.xml (hunk 1 above)
Comment (pocmo) [reply components/feature/addons/src/main/res/layout/mozac_feature_addons_item.xml:21]: Yeah, lint doesn't seem to recognize width/height from the style. I'll check if there's an option to just suppress.

Comment (pocmo) [reply components/feature/addons/src/main/res/layout/mozac_feature_addons_item.xml]: I removed it again and replaced it with `tools:ignore="RequiredSize"`

**Pull request merged**